substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -227,3 +227,161 @@ fn validate_serial_number(input: &[u8]) -> bool {
         .iter()
         .all(|c| matches!(c, b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'-' | b'_'))
 }
+
+pub const ONE_THOUSAND: u128 = 1_000;
+pub const GIB: u128 = 1024 * 1024 * 1024;
+
+/// A resources capacity that countains HRU, SRU, CRU and MRU in integer values.
+#[derive(Encode, Decode, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct Resources<T: Config> {
```

Comment:
The idea is to replace the `Resources` struc in support/src/types.rs by this one

Reply:
`Resources` type is also used in `pallet-smart-contract`, I moved it to `support/src/types.rs` to have the smart contract pallet not depend on the tfgrid pallet

Reply:
I suggest you create a new type `ResourcesInput`, where the validation can reside on.

Reply:
Done!
I found a solution that follows your idea